File: 383-RansomNote/383-RansomNote.java
// Last updated: 7/31/2025, 5:34:53 PM
class Solution {
    public boolean canConstruct(String ransomNote, String magazine) {
        // int [] rans = new int[26];
        int [] mag = new int[26];
        for(int i=0; i<magazine.length(); i++){
            int a = magazine.charAt(i)-'a';
            mag[a]++;
        }
        for(int i=0; i<ransomNote.length(); i++){
            int a = ransomNote.charAt(i)-'a';
            mag[a]--;
            if(mag[a]<0){
                return false;
            }
        }
        return true;
        
        
    }
}

File: 383-RansomNote/383-RansomNote.py
# Last updated: 7/31/2025, 5:34:59 PM
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        if ransomNote in magazine:
            return True
        a=list(magazine)
        c=0
        for i in ransomNote:
            if i in a:
                c=c+1
                a.remove(i)
            else:
                return False
        return True